src/server/db/firebase.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2017 The boardgame.io Authors
+ *
+ * Use of this source code is governed by a MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+const LRU = require('lru-cache');
+const firebase = require('firebase');
+
+/**
+ * Firebase RTDB/Firestore connector.
+ */
+export class Firebase {
+  /**
+   * Creates a new Firebase connector object.
+   */
+  constructor({ config, dbname, cacheSize, mockFirebase }) {
+    if (cacheSize === undefined) cacheSize = 1000;
+    if (dbname === undefined) dbname = 'bgio';
+    // TODO: better handling for possible errors
+    if (config === undefined) config = {};
+
+    this.mockFirebase = mockFirebase;
+    this.client = firebase;
+    // Default engine is Firestore
+    this.engine = config.engine === 'RTDB' ? config.engine : 'Firestore';
+    const { apiKey, authDomain, databaseURL, projectId } = config;
+    this.config = { apiKey, authDomain, databaseURL, projectId };
+    this.dbname = dbname;
+    this.cache = new LRU({ max: cacheSize });
+  }
+  /**
+   * Connect to the instance.
+   */
+  async connect() {
+    if (this.mockFirebase) {
```

Comment:
Instead of polluting the code with testing related stuff, how about doing:

```
db.db = mockFirebase;
```

in the test?

Reply:
(I think that should also help bring the test coverage up to 100% by executing lines 41 and 42).